### PR TITLE
Adding installIfNew.r example

### PR DESCRIPTION
**Pull request by 1beb**
Take a look, let me know what you think. I created a new file. It is essentially a copy of `install2.r` with some support for looking at `installed.packages()`. I added a warning when no packages will be installed but I'm not actually sure how that will be handled on passthrough.

### CODE REVIEW

File: inst/examples/installIfNew.R [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+      catch <-
+        grepl("package.*(is|are) not available", e$message) ||
+        grepl("installation of package.*had non-zero exit status", e$message)
+      if (catch) {
+        e <<- e
+      }
+    }
+  }
+
+  pkgs <- setdiff(pkgs, installed.packages()[,1])
```
Comment (1beb) [inline inst/examples/installIfNew.R]: Non trivial changes are here to L77

**Comment by eddelbuettel** [comment]
Let me take a closer peek at it in the next few days. 

First comment, confirmed from running after glancing at the code: `warning()` is noise.  Maybe just `message()` or even `cat()` ?

```r
edd@rob:~/git/littler/inst/examples(pr59/install_if_new)$ ./installIfNew.R RcppGSL
Warning message:
In install_packages2(pkgs = f, lib = lib, repos = if (isMatchingFile(f)) NULL else rep,  :
  All specified packages are already installed.
edd@rob:~/git/littler/inst/examples(pr59/install_if_new)$ 
```

We may want to simplify and remove pieces from `install2.r` we do not need, like the `-e` support. I forget what that was motivated by, once.

**Comment by eddelbuettel** [comment]
It is a pretty small change to `install2.r`.  Do you think it could / should just be an additional toggle there?

**Comment by eddelbuettel** [comment]
@1beb Any follow-up comment?  As small as it is, should it be over in `install2.r` ?  Default behaviour as is, if a (new) flag (say, `--skip-installed` or something) is set it does what you added.

**Comment by 1beb** [comment]
Sorry Dirk, I'm in transit will review this next week. I will program it as
you prefer it. It's really not that hard. So specify what your preference
is and I'll push an update.

**Comment by eddelbuettel** [comment]
In order to get on with life I am going to merge this, and then delete the new file, carrying the diff over to `install2.r` under an appropriate flag.

**Comment by 1beb** [comment]
Thx. I'm in Mexico at a Bachelor party. There is no programming this week.

**Pull request merged**